### PR TITLE
add dark mode and systems for color-scheme and theme

### PR DESCRIPTION
**Pull request by ryanatkn**
See the changelog for the specifics.

Adds the first pass at dark mode support. This meant adding systems for `color-scheme` and our own new concept of themes, which are named collections of variables.

Adds `$lib/theme.css` that's generated from the `'base'` theme. It's a simple transformation from the theme's variables array to CSS. Also adds themes `'high contrast'` and `'low contrast'` builtin themes.

Adds the `ThemeVariable` type used by `Theme`. Each variable has optional light and dark variants for the color schemes. This is now data that the user will be able to interact with in the live system. You can see a preview of this by clicking `"create a new theme"` or the `"..."` next to the builtin themes at `/library/theme`

Design goals:

- support building UIs with both the lightmode and darkmode perspectives, and prefer the darkmode way of looking at things - there's this idea of "adding light" and strategy of arbitrary alpha layering that it carries (reminds me of Harris' transitional design)
- use the OS default unless there's a user preference in `localStorage`, fall back to light, avoid flash on startup
- support custom themes
- base the systems on CSS vars
- enable scoping themes to parts of the page
- have a user-friendly correspondence between CSS vars and classes (direct mapping from `.a` to `--a` when it makes sense), be fairly restrained with adding classes
- do not enable scoping dark/light color schemes to parts of the page (it would complicate the code, and we don't see a compelling end-user usecase for this)
- expose an API for users to customize things
- make usage trivial -- see `Themed` and its top-level usage: https://github.com/feltjs/felt-ui/pull/247/files#diff-cc69d6a97e4f62578028d872b8d5032f2786fe1eaa65735429286b375234168d

References:

- https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme
- https://web.dev/prefers-color-scheme/
- https://web.dev/color-scheme/
- dark mode implementations that I took some ideas from or looked at:
  - Svelte - https://github.com/sveltejs/sites
  - Skeleton - https://github.com/skeletonlabs/skeleton
  - Tailwind - https://tailwindcss.com/
  - Mantine - https://mantine.dev/

Followup:

- hue input padding
- rename `font_X` to `size_X`, probably
- border colors might need alpha to not look broken
- button bgs for inputs (checkbox active for one)
- make the var detail dialog function to edit the var
- add library entry for `Themed` and `ColorSchemeInput`
- replace all of prose comments `/* allow nesting in .prose */` with `.prose h1:not(.unprose)` or `.unprose h1 {` or `.prose :not(.unprose) h1` (does the last one even work?)
- look at button bgs
- export `ThemeInput.svelte` (maybe refactor to events?)
- refactor storage into helpers (upstream to `@feltjs/util/storage.js`?)
- add interface to clear storage
- switch theme and color-scheme from global contextmenu
- integrate our colors the the Prism.js theme
- expose UI to change the tint
- How can we enable the value to be set from localStorage or a server-persisted account setting?
- maybe try to support multiple onscreen and scope to an element - would need to refactor to attach to a specific element, not the window
- I saw Pollen, another CSS framework that focuses on CSS vars, in a search after finishing up this PR, I'll look into how it does things

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -314,7 +340,7 @@
   ([#206](https://github.com/feltjs/felt-ui/pull/206))
 - add `util/dom.ts` helper `swallow` for stopping events concisely
   ([#208](https://github.com/feltjs/felt-ui/pull/208))
-- fix `.centered-hz` to explicitly set row direction
+- fix `.centered-h` to explicitly set row direction
```
Comment (ryanatkn) [inline changelog.md:343]:
```suggestion
- fix `.centered-hz` to explicitly set row direction
```

File: src/lib/style.css [hunk 2]
```diff
@@ -148,42 +104,41 @@
...
+	--spacing_lg: 1.3rem;
+	--spacing_xl: 1.6rem;
+	--spacing_xl2: 2.1rem;
+	--spacing_xl3: 2.6rem;
+	--spacing_xl4: 3.3rem;
+	--spacing_xl5: 4.2rem;
+	--spacing_xl6: 5.4rem;
+	--spacing_xl7: 6.9rem;
+	--spacing_xl8: 8.7rem;
+	--spacing_xl9: 11rem;
```
Comment (ryanatkn) [inline src/lib/style.css:132]:
```suggestion
	--spacing_xl9: 11.1rem;
```

File: changelog.md [hunk 3]
```diff
@@ -1,5 +1,37 @@
 # changelog
 
+## 0.55.0
+
+- **break**: add support for dark mode
+  ([#247](https://github.com/feltcoop/felt/pull/247))
+  - add `$lib/theme` with `createThemeSetupScript` for setting up the color scheme
+  - rename `.prose` from `.prose`
```
Comment (ryanatkn) [inline changelog.md:8]:
```suggestion
  - rename `.prose` from `.markup`
```

File: src/lib/ColorSchemeInput.svelte [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+		swallow(e);
+		$colorScheme = value;
+	};
+</script>
+
+<menu class="color-scheme-control">
+	{#each colorSchemes as c (c)}
+		<li role="none">
+			<button
+				class="menu_item"
```
Comment (ryanatkn) [inline src/lib/ColorSchemeInput.svelte:23]: type=button here any elsewhere

File: src/routes/+layout.svelte [hunk 5]
```diff
@@ -1,20 +1,34 @@
...
-	import Nav from '$lib/app/Nav.svelte';
+	import Nav from '$routes/Nav.svelte';
+	import {loadTheme} from '$lib/theme';
+	import Themed from '$lib/Themed.svelte';
+	import {DEFAULT_THEME, defaultThemes} from '$lib/themes';
 
 	const devmode = setDevmode(false);
+
+	const loadedTheme = loadTheme();
+	// TODO hacky
```
Comment (ryanatkn) [inline src/routes/+layout.svelte:16]: this hack will be removed when we add runtime theme editing

File: src/lib/themes.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
+import type {Theme} from '$lib/theme';
+
+export const DEFAULT_THEME_NAME = 'base';
+
+export const DEFAULT_THEME: Theme = {
+	name: DEFAULT_THEME_NAME,
+	variables: [], // see `variables` below -- base is the default set of variables
```
Comment (ryanatkn) [inline src/lib/themes.ts:7]:
```suggestion
	variables: [], // inherits base
```

File: src/lib/ColorSchemeInput.svelte (hunk 4 above)
Comment (greatbacon) [reply src/lib/ColorSchemeInput.svelte:29]: looks like this was resolved?

**Review by greatbacon — APPROVED** [review]
🍾

**Review by greatbacon — APPROVED** [review]
🍾

**Pull request merged**